practice/jokes/jokes/models.py

```diff
@@ -0,0 +1,87 @@
+"""jokes Models."""
+
+
+class Joke:
+    """Value type that holds a joke's setup and punchline.
+
+    Both setup and punchline are text.
+    """
+    def __init__(self, setup, punchline):
+        r""" Initialize new Joke.
+
+        >>> a = Joke('knock-knock', 'who 1 dat?')
+        >>> a.setup
+        'knock-knock'
+        >>> a.punchline
+        'who 1 dat?'
+        """
+        self.setup = setup
+        self.punchline = punchline
+
+    def __eq__(self, other):
+        r""" Test for equality.
+
+        >>> a = Joke('knock-knock', 'who dat?')
+        >>> b = Joke('blah blah blah', 'blah!')
+        >>> a == a
+        True
+        >>> a == b
+        False
+        """
+        return self.setup == other.setup and self.punchline == other.punchline
+
+    def __repr__(self):
+        r""" Magic repr function.
+
+        >>> Joke('knock-knock', 'who dat?')
+        Joke(knock-knock, who dat?)
+        """
+        return 'Joke({}, {})'.format(self.setup, self.punchline)
+
+
+_JOKES = []
+
+
+def get_all_jokes():
+    r"""Return all of the jokes.
+
+    >>> get_all_jokes()
+    [['knock-knock', 'who 2 dat?'], ['bahbah', 'blacksheep']]
+    """
+    return [
+        [joke.setup, joke.punchline]
```

Comment:
What is the purpose of unpacking the Joke class instances? In the rest of your code you can use them, including the templates. It's okay for a model class to be used in the template.